checker/src/main/java/org/checkerframework/checker/nullness/NullnessNoInitAnnotatedTypeFactory.java

```diff
@@ -483,6 +486,19 @@ protected void replacePolyQualifier(AnnotatedTypeMirror lhsType, Tree context) {
         }
     }
 
+    @Override
+    protected void addComputedTypeAnnotations(Tree tree, AnnotatedTypeMirror type) {
+        if (tree != null && tree instanceof NewClassTree) {
+            String treeKey = tree.toString();
```

Comment:
Serializing a Tree can be very costly - think of an anonymous class that can have a complicated body.
Why is that a good key? Can't you just use the object reference?